homework/index.js

```diff
@@ -30,14 +30,120 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const listContainer = document.getElementById('list-container');
+    clearContainer(listContainer);
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert alert-error' });
+  }
+
+  function addRow(tbody, label, repository) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    if (label === 'Repository:') {
+      const urlRepo = repository.html_url;
+      const td = createAndAppend('td', tr);
+      createAndAppend('a', td, {
+        text: repository.name,
+        href: urlRepo,
+        target: '_blank',
+      });
+    }
+    if (label === 'Description:') {
+      createAndAppend('td', tr, { text: repository.description });
+      console.log(repository.description);
+    }
+    if (label === 'Forks:') {
+      createAndAppend('td', tr, { text: repository.forks });
+    }
+
+    if (label === 'Updated:') {
+      createAndAppend('td', tr, { text: repository.updated_at });
+    }
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const div = createAndAppend('div', container, { id: 'info_div' });
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    createAndAppend('tr', table);
+    createAndAppend('tr', table, {});
+    addRow(tbody, 'Repository:', repository);
+    addRow(tbody, 'Description:', repository);
+    addRow(tbody, 'Forks:', repository);
+    addRow(tbody, 'Updated:', repository);
+  }
+
+  function createContributorsList(repositoryName, ul) {
+    clearContainer(ul);
+    const url = `https://api.github.com/repos/HackYourFuture/${repositoryName}/contributors`;
+    fetchJSON(url, (err, data) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+
+      createAndAppend('h2', ul, {
+        text: 'contributions',
+      });
+      data.map(item => {
+        const li = createAndAppend('li', ul);
+        const div = createAndAppend('div', li);
+
+        createAndAppend('img', div, {
+          class: 'contributor_avatar',
+          src: item.avatar_url,
+          alt: item.login,
+        });
+        const p1 = createAndAppend('p', div, { id: 'p1' });
+        const p2 = createAndAppend('p', div, { id: 'p2' });
+        p1.textContent = item.login;
```

Comment:
1. It is not necessary to use `.textContent` or `.innerHTML` properties here. Instead, you can use the `text` property of `createAndAppend()`. As a bonus it also allows you to get rid of the `p1` and `p2` variables.

    ```js
    createAndAppend('p', div, { id: 'p1', text: item.login });
    createAndAppend('p', div, { id: 'p2', text: item.contributions });
    ```

2. This assignment requires that you should be able to visit the contributors GitHub page in a new tab when you click on the list item (or navigate to it with the tab key and press Enter, for visually impaired users).